pkg/util/cluster/cluster.go

```diff
@@ -589,15 +590,15 @@ func (c *Cluster) fixupNSGs(ctx context.Context, vnetResourceGroup, clusterName
 }
 
 func (c *Cluster) deleteRoleAssignments(ctx context.Context, vnetResourceGroup, appID string) error {
-	spObjID, err := c.getServicePrincipal(ctx, appID)
+	spObjID, err := graphutil.GetServicePrincipalIDByAppID(ctx, c.graph, appID)
 	if err != nil {
 		return err
 	}
-	if spObjID == "" {
+	if spObjID == nil {
 		return nil
 	}
 
-	roleAssignments, err := c.roleassignments.ListForResourceGroup(ctx, vnetResourceGroup, fmt.Sprintf("principalId eq '%s'", spObjID))
+	roleAssignments, err := c.roleassignments.ListForResourceGroup(ctx, vnetResourceGroup, fmt.Sprintf("principalId eq '%s'", *spObjID))
```

Comment:
This is the use case that led me to have `GetServicePrincipalIdByAppID` return `nil` for not found instead of an error.

`deleteRoleAssignments` is idempotent, so if it can't find the app ID then it returns no error.  Checking for `nil` here just seemed simpler than parsing an `error`, but if you still feel it should eat an error here then I can revise.

Reply:
I'll admit it is a stylistic choice, so I'll let others comment too - I'd hold off on the change and see what others think. IMO, if we have 1 case where a "not found" is expected, and all others without, we could do a specific error handling check on that 1 case to achieve the same result (if the error is a `404` error, then...).

Reply:
Note, however, there's only 3 cases total.  But I'm willing to go either way based on consensus.

Reply:
I am in favor of not using nil as a 'happy path' value when we can't find the service principal  when given an appId this should be an error (handled) but not expected.

Reply:
>but not expected

I wouldn't return an error, but log a message stating no role assignments found.  

If the cluster installation fails or is requeued for any reason, this role assignment may not exist.  And that's totally fine, we're just trying to clean up after ourselves on cluster deletion.

Reply:
I think this discussion thread shifted slightly.

Originally we were debating whether `GetServicePrincipalIDByAppID` should return an error if no service principals are found.  It was a stylistic choice on my part to just return a `nil` string pointer.

What I'm understanding from the last couple comments is a "not found" error from `c.roleassignments.ListForResourceGroup` should simply be logged?  (If it even returns an error in that case, I'm not sure if it does.)  I'll investigate and open a followup PR if necessary, if that's alright.